viewer/api/models.py

```diff
@@ -24,6 +24,12 @@ def __unicode__(self):
         return u'%s, date=%s, display=%s' % (
             self.name, self.date.strftime('%Y-%m-%d'), self.display)
 
+    def get_metrics(self):
+        metrics = set()
+        for collection in Collection.objects.filter(dataset=self):
+            metrics.add(collection.metric.id)
+        return list(metrics)
```

Comment:
This could all be done at the database level in a single query:
`Metric.objects.filter(collection__dataset=self).distinct('id').values_list('id', flat=True)`